android/src/main/java/com/tailscale/ipn/ui/localapi/LocalAPIRequest.kt

```diff
@@ -41,59 +47,41 @@ private object Endpoint {
 //
 // (jonathan) TODO: Audit local API for all of the possible error results and clean
 // it up if possible.
-enum class APIErrorVals(val rawValue: String) {
-    UNPARSEABLE_RESPONSE("Unparseable localAPI response"),
-    NOT_READY("Not Ready");
+enum class APIErrorVals(private val rawValue: String) {
+    UNPARSEABLE_RESPONSE("Unparseable localAPI response"), NOT_READY("Not Ready");
 
     fun toError(): Error {
         return Error(rawValue)
     }
 }
 
 class LocalAPIRequest<T>(
-        path: String,
-        val method: String,
-        val body: String? = null,
-        val parser: (String) -> Unit,
+    path: String,
+    val method: String,
+    val body: ByteArray? = null,
+    val parser: (ByteArray) -> Unit,
 ) {
     val path = "/localapi/v0/$path"
+    val cookie = UUID.randomUUID().toString()
```

Comment:
Using a UUID eliminates the need for locking. In the past, I've run into weird issues combining Kotlin coroutines and Java locks, so whenever we can avoid the `synchronized` keyword, it's probably better to do so.